sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/CharVarcharUtils.scala

```diff
@@ -161,9 +162,20 @@ object CharVarcharUtils extends Logging {
 
   private def paddingWithLengthCheck(expr: Expression, dt: DataType): Expression = dt match {
     case CharType(length) =>
-      StringRPad(stringLengthCheck(expr, dt, needTrim = false), Literal(length))
-
-    case VarcharType(_) => stringLengthCheck(expr, dt, needTrim = false)
+      StaticInvoke(
```

Comment:
`StringRPad`'s gen-ed code is pretty short, as `s"$str.rpad($len, $pad)"`. Does `StaticInvoke` help reduce gen-ed code?

Reply:
It not just a single rad but,

StringRPad(If(GreaterThan(Length(StringTrimRight(expr)), Literal(length)), raiseError("char", length),StringTrimRight(expr))))

Reply:
Oh I see. Then we may need a benchmark to show the difference.